synapse/handlers/relations.py

```diff
@@ -453,19 +405,43 @@ async def get_bundled_aggregations(
             latest_thread_event = thread.latest_event
             if latest_thread_event and latest_thread_event.event_id not in events_by_id:
                 events_by_id[latest_thread_event.event_id] = latest_thread_event
+                # The latest event in the thread must have a thread relation.
+                relations_by_id[latest_thread_event.event_id] = RelationTypes.THREAD
 
         # Fetch other relations per event.
         for event in events_by_id.values():
-            (
-                annotations,
-                references,
-            ) = await self._get_bundled_annotations_and_references_for_event(
-                event, ignored_users
+            # Edits and annotations may not have related annotations or references.
```

Comment:
This comment was originally added in #11284 (and then has moved around quite a bit...); it originally comes from a comment in the REST layer:

https://github.com/matrix-org/synapse/blob/6a5dd485bd82b269e7e169c0385290d081eae801/synapse/rest/client/relations.py#L233-L235

As part of #11284 I seem to have made it so that this behavior applies only to annotations (reactions) and references (replies).

I think the following might be clearer:

> It doesn't make sense for an annotation (ie, reaction) or reference (ie, reply) to have any other events related to them.

The original comment seems confusing that it talks about edits though, I suspect that was a typo. There was a thread on #11284 which talks about this: https://github.com/matrix-org/synapse/pull/11284#discussion_r754362364

Note that [MSC2677](https://github.com/uhoreg/matrix-doc/blob/aggregations-reactions/proposals/2677-reactions.md#edge-cases) says:

> Can you react to a reaction?
> * Yes, at the protocol level. But you shouldn't expect clients to do anything useful with it.

...but it doesn't say anything about whether those should be bundled. Maybe this check is just bogus here and we should remove it?